examples/characteristics_advection.cpp

```diff
@@ -0,0 +1,328 @@
+    auto const y_domain = ddc::init_discrete_space(
+            DDimY::
+                    init(ddc::Coordinate<Y>(y_start),
+                         ddc::Coordinate<Y>(y_end),
+                         ddc::DiscreteVector<DDimY>(nb_y_points)));
+
+    //! [time-domains]
+    // max(1/dx^2)
+    double const invdx2_max = ddc::transform_reduce(
+            x_domain,
+            0.,
+            ddc::reducer::max<double>(),
+            [](ddc::DiscreteElement<DDimX> ix) {
+                return 1.
+                       / (ddc::distance_at_left(ix)
+                          * ddc::distance_at_right(ix));
+            });
+    // max(1/dy^2)
+    double const invdy2_max = ddc::transform_reduce(
+            y_domain,
+            0.,
+            ddc::reducer::max<double>(),
+            [](ddc::DiscreteElement<DDimY> iy) {
+                return 1.
+                       / (ddc::distance_at_left(iy)
+                          * ddc::distance_at_right(iy));
+            });
+    ddc::Coordinate<T> const max_dt {0.1};
+
+    // number of time intervals required to reach the end time
+    ddc::DiscreteVector<DDimT> const nb_time_steps {
+            std::ceil((end_time - start_time) / max_dt) + .2};
+    // Initialization of the global domain in time:
+    // - the number of discrete time-points is equal to the number of
+    //   steps + 1
+    ddc::DiscreteDomain<DDimT> const time_domain
+            = ddc::init_discrete_space(
+                    DDimT::
+                            init(ddc::Coordinate<T>(start_time),
+                                 ddc::Coordinate<T>(end_time),
+                                 nb_time_steps + 1));
+    //! [time-domains]
+
+    //! [data allocation]
+    // Maps density into the full domain twice:
+    // - once for the last fully computed time-step
+    ddc::Chunk _last_density(
+            ddc::DiscreteDomain<DDimX, DDimY>(x_domain, y_domain),
+            ddc::DeviceAllocator<double>());
+
+    // - once for time-step being computed
+    ddc::Chunk _next_density(
+            ddc::DiscreteDomain<DDimX, DDimY>(x_domain, y_domain),
+            ddc::DeviceAllocator<double>());
+    //! [data allocation]
+
+    //! [initial-conditions]
+    ddc::ChunkSpan const initial_density = _last_density.span_view();
+    // Initialize the density on the main domain
+    ddc::DiscreteDomain<DDimX, DDimY> x_mesh
+            = ddc::DiscreteDomain<DDimX, DDimY>(x_domain, y_domain);
+    ddc::for_each(
+            ddc::policies::parallel_device,
+            x_mesh,
+            DDC_LAMBDA(ddc::DiscreteElement<DDimX, DDimY> const ixy) {
+                double const x
+                        = ddc::coordinate(ddc::select<DDimX>(ixy));
+                double const y
+                        = ddc::coordinate(ddc::select<DDimY>(ixy));
+                // initial_density(ixy) = 9.999 * Kokkos::exp(-(x * x + y * y) / 0.1 / 2);
+                initial_density(ixy) = 9.999 * ((x * x + y * y) < 0.25);
```

Comment:
Looks quite discontinuous to be precisely represented by splines. Let's use the gaussian function.
```suggestion
                initial_density(ixy) = 9.999 * Kokkos::exp(-(x * x + y * y) / 0.1 / 2);
```

Reply:
I like to see distorsion appearing on the discontinuity and integral staying unchanged despite of it, but as you prefer